src/doc/src/reference/config.md

```diff
@@ -828,6 +828,57 @@ without pipelining.
 Specifies a custom user-agent header to use. The default if not specified is a
 string that includes Cargo's version.
 
+### `include`
+
+* Type: string, array of strings, or array of tables
+* Default: none
+* Environment: not supported
+
+Loads additional config files. Paths are relative to the config file that
+includes them. Only paths ending with `.toml` are accepted.
+
+Supports the following formats:
+
+```toml
+# single path
+include = "path/to/mordor.toml"
+
+# array of paths
+include = ["frodo.toml", "samwise.toml"]
```

Comment:
Are there other cases where we allow for singular and plural versions of fields?

I know I've run into problems forgetting `[]` with `features` in dependencies...

Reply:
`--target` and `--package`, and `--features` CLI flags are also inconsistent.
Probably slightly prefer to stabilize with `include` to have less churn on nightly users (not a strong argument though).

We can add some warnings for all the singular/plural confusion for configs. Something similar to this <https://github.com/rust-lang/cargo/pull/16256> but instead check the ConfigValue hashmap.